cli/internal/core/engine.go

```diff
@@ -142,10 +148,36 @@ func (e *Engine) generateTaskGraph(pkgs []string, taskNames []string, tasksOnly
 		traversalQueue = traversalQueue[1:]
 
 		pkg, taskName := util.GetPackageTaskFromId(taskID)
+
+		// This for lopo adds in the Root Node as a taskID (i.e. __ROOT__#build)
+		// into the traversalQueue. But in order to look up the information about
+		// the root workpsace, we need to use the package name "//". Re-assign
+		// the pkgName here to account for that.
+		// TODO(mehulkar): Is it safe to move this logic into GetPackageTaskFromId()?
+		if pkg == ROOT_NODE_NAME {
+			pkg = util.RootPkgName
+		}
```

Comment:
I'm not sure if this check should instead be to _skip_ the __ROOT_ node instead? All tests are passing, so it doesn't seem like this is a problem, but in other places, we have checks to skip this root node.

Reply:
I _think_ it would be hit some times. there's a difference between a `//#build` and `__ROOT__#build` task. The first is user-configured, the second is some virtual node that we insert into the graph (and I can't remember why). I think this rewrite might actually be problematic because the next condition will be exercised for the virtual node also. 

I'll spend some more time understanding this / adding a test case in.

Reply:
I was actually able to remove this piece of code. It was left over from doing actual composition in #2706, but in this refactor it's not necessary, so I can delay doing this work.